aspnetcore/web-api/define-controller/samples/WebApiSample.Api.Pre21/Controllers/OrdersController.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System.Threading.Tasks;
+using WebApiSample.DataAccess.Models;
+using WebApiSample.DataAccess.Repositories;
+
+namespace WebApiSample.Api.Controllers
+{
+    #region snippet_OrdersController
+    public class OrdersController : Controller
+    {
+        private readonly OrdersRepository _repository;
+
+        public OrdersController(OrdersRepository repository)
+        {
+            _repository = repository;
+        }
+
+        public IActionResult Index() => View();
+
+        [HttpGet("{id}")]
+        [ProducesResponseType(typeof(Order), 200)]
+        [ProducesResponseType(404)]
+        public IActionResult GetById(int id)
```

Comment:
This should use `ActionResult<Order>`

Reply:
That allows you to remove the `[ProducesResponseType]` attribute when it's a 200.

Reply:
This was done intentionally to show how things are accomplished prior to 2.1.

Reply:
This is a 2.0 version of the sample app, which is why it doesn't use ActionResult<T>.